src/drivers/apl-ssp.c

```diff
@@ -549,7 +549,13 @@ static inline int ssp_set_config(struct dai *dai,
 		sscr0 |= SSCR0_MOD | SSCR0_FRDC(config->ssp.tdm_slots);
 
 		/* set asserted frame length */
-		frame_len = 1;
+		frame_len = 1; /* default */
+
+		if (ssp->params.frame_pulse_width &
```

Comment:
It's a bit odd.
I would have done something like

if (cfs && ssp->params.frame_pulse_width > 0 && ssp-params.frame_pulse_width < MAX)

Reply:
@plbossart By cfs do you mean SOF_DAI_FMT_CBS_CFS format?

Reply:
Both SOF_DAI_FMT_CBS_CFS and SOF_DAI_FMT_CBM_CFS.
cfs is the variable I used for byt-ssp, we should do the same in apl-ssp.c and only set what is meaningful for the SoC frame master mode.